sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -1615,7 +1615,7 @@ def stage_block(
             length=length,
             **kwargs)
         try:
-            self._client.block_blob.stage_block(**options)
+            return self._client.block_blob.stage_block(**options)
```

Comment:
We'd also want to propagate these changes to async. ( `storage/blob/aio/_blob_client.py` ) .

Also, we'd want to add `'cls': return_response_headers,` to _stage_block_options (line 1559). That way there will be no need to create a ResponseHandler class if the goal is to just get the headers.

Reply:
Currently, the default is to return `None`. This would change the default behavior. I'm happy to go either way on this. Are you suggesting that the default behavior should change? For now, I used a function instead of the class to get the response and headers.

Reply:
Added the corresponding change to `aio/_blob_client.py`

Reply:
Actually, on a second thought, lets not change the current behavior.

Reply:
Hi @timuralp 
Thanks for making this change!
Would you like to also help teak the doc a bit along with this PR ^_^
1. for line #1582 would you like to make it from `# type: (...) -> None` to `# type: (...) -> Dict[str, Any]`
2. for line #1613 would you like to make it
```
:returns: Blob property dict
:rtype: dict[str, Any]
```

We are really appreciated for your help on this!

Reply:
Done